src/Oscoin/Data/OscoinTx.hs

```diff
@@ -31,15 +31,15 @@ type Tx c = Tx' 1 0 c
 
 -- | A transaction with parametric version and flags.
 data Tx' (version :: Nat) (flags :: Nat) c = Tx'
-    { txPayload :: TxPayload c
+    { txMessages :: [TxMessage c]
```

Comment:
Should be `Set (TxMessage c)`, I would say.

Reply:
Hmm you might be right, including the same message twice would more likely be an error than deliberate.

Reply:
Another issue is that the size of the list should be restricted.

Reply:
> Another issue is that the size of the list should be restricted.

Thought about this, but I think it's easiest to let this be handled by tx byte-size validation (which is a todo in the code). We could impose a limit on the list length too, but it seems redundant if we have a limit on tx size. We're also going to be adjusting the tx fee based on the list length.